tests/js/setup.ts

```diff
@@ -219,6 +211,92 @@ const routerFixtures = {
 };
 
 type TestStubTypes = TestStubFixtures & typeof routerFixtures;
+const jsFixturesDirectory = path.resolve(__dirname, '../../fixtures/js-stubs/');
+
+const extensions = ['.js', '.ts', '.tsx', '.json'];
+
+const specialMapping = {
+  AllAuthenticators: 'authenticators.js',
+  OrgRoleList: 'roleList.js',
+  MetricsField: 'metrics.js',
+  EventsStats: 'events.js',
+  DetailedEvents: 'events.js',
+  Events: 'events.js',
+  OutcomesWithReason: 'outcomes.js',
+  SentryAppComponentAsync: 'sentryAppComponent.js',
+  EventStacktraceMessage: 'eventStacktraceException.js',
+  MetricsTotalCountByReleaseIn24h: 'metrics.js',
+  MetricsSessionUserCountByStatusByRelease: 'metrics.js',
+  MOCK_RESP_VERBOSE: 'ruleConditions.js',
+  SessionStatusCountByProjectInPeriod: 'sessions.js',
+  SessionUserCountByStatusByRelease: 'sessions.js',
+  SessionUserCountByStatus: 'sessions.js',
+  SessionStatusCountByReleaseInPeriod: 'sessions.js',
+  SessionsField: 'sessions.js',
+  ProviderList: 'integrationListDirectory.js',
+  BitbucketIntegrationConfig: 'integrationListDirectory.js',
+  GitHubIntegration: 'githubIntegration.js',
+  GitHubRepositoryProvider: 'githubRepositoryProvider.js',
+  GitHubIntegrationProvider: 'githubIntegrationProvider.js',
+  GitHubIntegrationConfig: 'integrationListDirectory.js',
+  OrgOwnedApps: 'integrationListDirectory.js',
+  PublishedApps: 'integrationListDirectory.js',
+  SentryAppInstalls: 'integrationListDirectory.js',
+  PluginListConfig: 'integrationListDirectory.js',
+  DiscoverSavedQuery: 'discover.js',
+  VercelProvider: 'vercelIntegration.js',
+  TagValues: 'tagvalues.js',
+};
+
+function tryRequire(dir: string, name: string): any {
+  if (specialMapping[name]) {
+    return require(path.resolve(dir, specialMapping[name]));
+  }
+  for (const ext of extensions) {
+    try {
+      return require(path.resolve(dir, lowercaseFirst(name) + ext));
+    } catch {
+      // ignore
+    }
+  }
+  throw new Error('Failed to resolve file');
+}
+
+function lowercaseFirst(value: string): string {
+  return value.charAt(0).toLowerCase() + value.slice(1);
+}
+
+const lazyFixtures = new Proxy(
+  {},
+  {
+    get(target, prop: string) {
+      if (target[prop]) {
+        return target[prop];
+      }
+      if (routerFixtures[prop]) {
+        return routerFixtures[prop];
+      }
+
+      try {
+        const maybeModule = tryRequire(jsFixturesDirectory, prop);
+        for (const exportKey in maybeModule) {
+          target[exportKey] = maybeModule[exportKey];
+        }
+      } catch {
+        // ignore
+      }
+
+      if (target[prop] === undefined) {
+        return () => {
+          throw new Error(`Fixture ${prop} does not exist`);
+        };
+      }
+      return target[prop];
+    },
+  }
+) as TestStubTypes;
+
+ConfigStore.loadInitialData(lazyFixtures.Config());
```

Comment:
This all looks great. Can we maybe move all of this out into a separate file / module though? the test setup is already longer than it should be :(

Reply:
Yeah, I'll do that and improve the error message so that if folks have missing stubs they know where to look to.

Reply:
Great stuff 👍 

the fixtures / stubs are a pretty old mechanism that definitely can use some love

Reply:
@evanpurkhiser we should discuss this at a TSC, but one thing I tried was to import them directly in a test and TS was screaming with a lot of errors. I'm worried these are falling out of sync with the actual types and giving us some false safety

Reply:
Yeah that sounds right.

I think once we get off of enzyme the next thing we should maybe turn our attention to is typescript for our tests to avoid the situation you're describing

Reply:
> @evanpurkhiser we should discuss this at a TSC, but one thing I tried was to import them directly in a test and TS was screaming with a lot of errors. I'm worried these are falling out of sync with the actual types and giving us some false safety

Agree that this is a good topic for TSC. We should definitely discuss this :)

Reply:
> I think once we get off of enzyme the next thing we should maybe turn our attention to is typescript for our tests to avoid the situation you're describing

agree @evanpurkhiser . can't wait for having all our tests and stubs typed too! a dream 🙂